src/contractor/contractor.cpp

```diff
@@ -81,6 +81,38 @@ inline EdgeWeight distanceAndSpeedToWeight(double distance_in_meters, double spe
     return std::max<EdgeWeight>(1, static_cast<EdgeWeight>(std::round(duration * 10)));
 }
 
+// Returns updated edge weight
+template <class IterType>
+EdgeWeight getNewWeight(IterType new_speed_iter,
+                        const double &segment_length,
+                        const std::vector<std::string> &segment_speed_filenames,
+                        EdgeWeight oldWeight,
+                        const double &verify_weights)
+{
+    auto new_segment_weight =
+        (new_speed_iter->speed_source.speed > 0)
+            ? distanceAndSpeedToWeight(segment_length, new_speed_iter->speed_source.speed)
+            : INVALID_EDGE_WEIGHT;
+    // the check here is enabled by the `--verify-weights` flag
+    // it logs a warning if the new weight exceeds a heuristic of what a reasonable weight update is
+    if (verify_weights > 0)
+    {
+        if ((new_segment_weight < oldWeight) && ((oldWeight / new_segment_weight) > verify_weights))
```

Comment:
hm. I don't think that we should rely on some outside check at another location. It might, for some reason, might be considered reasonable to set to zero some point down the line. If it isn't guaranteed here, we should check for it.